tools/privacyidea-decrypt-safeword.py

```diff
@@ -46,6 +46,13 @@
                     des = DES.new(des_key)
                     d = des.decrypt(binascii.unhexlify(data))
                     current_token["seed"] = d.strip("\x00")
+
+                counter = m.group(1)
+                algo, data = counter.split(":")
+                if algo.lower() == "des":
+                    des = DES.new(des_key)
```

Comment:
This requires the unsupported (since 2014) and undocumented PyCrypto library. Even the drop-in replacement PyCryptoDome requires a mode: https://www.pycryptodome.org/en/latest/src/cipher/des.html
According to https://stackoverflow.com/questions/54237266/des-cipher-with-pyca-cryptography-pbewithmd5anddes we can use the `cryptography` library, we just need to check which mode is used.

Reply:
I think this is a generic problem. We would also have to remove the AES decryption. Well, I think the AES decryption is not used anymore. The DES is used for SAM export ldif files. 
The AES will probably not be used anymore in the future, since new tokens are not delivered in ldif.
So this script is deprecated in itself anyways.

I will look into DES from cryptography, if this can *easily* help.

Reply:
Yes, without test data this is cumbersome.
But we need to be clear that this script does not work with the current environment.
We can add a `try...except ImportError` around the import of `Crypto` to catch this.

Reply:
Maybe we should move the script from the privacyidea-repo to the the scripts repo. Thus it would get a more "unofficial" character.

Reply:
I moved it here:
https://github.com/privacyidea/scripts/blob/master/toolbox/privacyidea-decrypt-safeword.py